src/imgui/Shortcuts.cc

```diff
@@ -0,0 +1,60 @@
+#include "Shortcuts.hh"
+#include "imgui_internal.h"
+
+#include "SettingsConfig.hh"
+
+#include <array>
+
+namespace openmsx {
+
+static constexpr auto defaultShortcuts = std::array{
+	Shortcuts::Data{ShortcutIndex::GOTO_MEMORY_ADDRESS, ImGuiMod_Ctrl | ImGuiKey_G, ShortcutType::LOCAL},
+	Shortcuts::Data{ShortcutIndex::GOTO_DISASM_ADDRESS, ImGuiMod_None, ShortcutType::LOCAL},
+};
+
+Shortcuts::Shortcuts(SettingsConfig& config_)
+	: config(config_)
+	, shortcuts{}
+{
+	config.setShortcuts(*this);
+	loadInit();
+}
+
+void Shortcuts::loadInit()
+{
+	for (auto& shortcut : defaultShortcuts) {
+		setShortcut(shortcut.index, shortcut.keychord, shortcut.local, shortcut.repeat);
+	}
+}
+
+Shortcuts::Data& Shortcuts::getShortcut(ShortcutIndex index)
+{
+	assert(index < ShortcutIndex::NUM_SHORTCUTS);
+	return shortcuts[index];
+}
+
+void Shortcuts::setShortcut(ShortcutIndex index, ImGuiKeyChord keychord, std::optional<bool> local, std::optional<bool> repeat)
```

Comment:
This is weird. It's always required to set `keychord`, but you can leave `local` and `repeat` unchanged.
E.g. what if (via the GUI) I only want to change local<->global.

Either make also `keyChord` optional. Or (my preference) make separate methods to change each field (`setKeychord()`, `setType()`).